src/metrics.ts

```diff
@@ -626,8 +700,29 @@ export function getMetrics(
     /** Register the inclusion of peers in our mesh due to some reason. */
     onAddToMesh(topicStr: TopicStr, reason: InclusionReason, count: number): void {
       const topic = this.toTopic(topicStr)
-      this.meshPeerInclusionEvents.inc({ reason }, count)
-      this.meshPeerInclusionEventsByTopic.inc({ topic }, count)
+      switch (reason) {
+        case InclusionReason.Fanout:
+          this.meshPeerInclusionEventsFanout.inc({ topic }, count)
+          break
+        case InclusionReason.Random:
+          this.meshPeerInclusionEventsRandom.inc({ topic }, count)
+          break
+        case InclusionReason.Subscribed:
+          this.meshPeerInclusionEventsSubscribed.inc({ topic }, count)
+          break
+        case InclusionReason.Outbound:
+          this.meshPeerInclusionEventsOutbound.inc({ topic }, count)
+          break
+        case InclusionReason.NotEnough:
+          this.meshPeerInclusionEventsNotEnough.inc({ topic }, count)
+          break
+        case InclusionReason.Opportunistic:
+          this.meshPeerInclusionEventsOpportunistic.inc({ topic }, count)
+          break
+        default:
+          this.meshPeerInclusionEventsUnknown.inc({ topic }, count)
+          break
```

Comment:
Is it really necessary to have so much granularity here?

Reply:
@dapplion  without it we don't know how mesh peers are added for `beacon_attestation` topic like before

maybe it's not important for this specific metric, but the same granularity for `ChurnReason` metrics below are quite important when investigating low mesh peer issue

so it's more code here but beneficial for the applications using gossipsub

Reply:
Okk sounds good